src/main/java/com/celements/cells/AbstractWriter.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.celements.cells;
+
+import java.util.Collections;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import com.celements.cells.attribute.CellAttribute;
+
+@NotThreadSafe
+public abstract class AbstractWriter implements ICellWriter {
```

Comment:
I don't think there is a need for this abstraction since JsonWriter doesn't use the openLevels deque

Reply:
`getOpenLevels` is part of the `ICellWriter` interface and this implementation looks to me like a sensible default implementation. Therefore I used the same implementation for `DivWriter` and `JsonWriter` and extracted it to a `AbstractWriter` class.